jbpm-recommendation-smile-random-forest/src/main/java/org/jbpm/prediction/randomforest/SmileRandomForest.java

```diff
@@ -0,0 +1,234 @@
+     *
+     * @return A map of input attributes with the attribute name as key and attribute type as value.
+     */
+    private static RandomForestConfiguration readConfigurationFromFile() {
+
+        final RandomForestConfiguration configuration = new RandomForestConfiguration();
+
+        InputStream inputStream = null;
+        final Map<String, AttributeType> inputFeatures = new HashMap<>();
+        try {
+            Properties prop = new Properties();
+
+            inputStream = SmileRandomForest.class.getClassLoader().getResourceAsStream("inputs.properties");
+
+            if (inputStream != null) {
+                prop.load(inputStream);
+            } else {
+                throw new FileNotFoundException("Could not find the property file 'inputs.properties' in the classpath.");
+            }
+
+            for (Object propertyName : prop.keySet()) {
+                inputFeatures.put((String) propertyName, AttributeType.valueOf(prop.getProperty((String) propertyName)));
+            }
+
+        } catch (Exception e) {
+            logger.error("Exception: " + e);
+        }
+        configuration.setInputFeatures(inputFeatures);
+
+        try {
+            Properties prop = new Properties();
+
+            inputStream = SmileRandomForest.class.getClassLoader().getResourceAsStream("output.properties");
+
+            if (inputStream != null) {
+                prop.load(inputStream);
+            } else {
+                throw new FileNotFoundException("Could not find the property file 'output.properties' in the classpath.");
+            }
+
+            configuration.setOutcomeName(prop.getProperty("name"));
+            configuration.setOutcomeType(AttributeType.valueOf(prop.getProperty("type")));
+            configuration.setConfidenceThreshold(Double.parseDouble(prop.getProperty("confidence_threshold")));
+            configuration.setNumTrees(Integer.parseInt(prop.getProperty("num_trees")));
+        } catch (Exception e) {
+            logger.error("Exception: " + e);
+        }
+        return configuration;
+    }
+
+    private static Attribute createAttribute(String name, AttributeType type) {
+        if (type == AttributeType.NOMINAL) {
+            return new NominalAttribute(name);
+        } else if (type == AttributeType.NUMERIC) {
+            return new NumericAttribute(name);
+        } else {
+            return new StringAttribute(name);
+        }
+    }
+
+    /**
+     * Add the data provided as a map to a Smile {@link smile.data.Dataset}.
+     *
+     * @param data    A map containing the input attribute names as keys and the attribute values as values.
+     * @param outcome The value of the outcome (output data).
+     */
+    public void addData(Map<String, Object> data, Object outcome) {
+        final double[] features = new double[numAttributes];
+        int i = 0;
+        for (String attrName : smileAttributes.keySet()) {
```

Comment:
Same as before: As key and value are needed during the iteration, it's more efficient to do it with entrySet
```suggestion
        for (Map.Entry<String, Attribute> attrName : smileAttributes.entrySet()) {
```